src/Core/src/Platform/iOS/ImageViewExtensions.cs

```diff
@@ -87,15 +87,15 @@ internal static CGSize SizeThatFitsImage(
 			double constrainedWidth = Math.Min(contentWidth, constraints.Width);
 			double constrainedHeight = Math.Min(contentHeight, constraints.Height);
 
-			double widthRatio = constrainedWidth / imageSize.Width;
-			double heightRatio = constrainedHeight / imageSize.Height;
+			double widthRatio = constrainedWidth / contentWidth;
+			double heightRatio = constrainedHeight / contentHeight;
```

Comment:
To confirm what I am thinking, this is the size used to determine the size of the control, so you are including the padding in here because iOS does not have a [padding concept?

But, as @PureWeen mentioned - should it be used in the aspect ratio calculations? Maybe it should be added on after the size calculations? For example, if I have a landscape image, but massive padding on top/bottom that makes it become a portrait scale. Will it cause issues with stretch?

Maybe I am thinking wrong here?

Reply:
Good call, I've updated the algorithm as follows:

- Remove padding from constraints
- Compute desired size on reduced constraints
- Add padding back